test/cpp/tensorexpr/test_memplanning.cpp

```diff
@@ -85,6 +85,195 @@ TEST(BufLiveRange, MulRangeLine) {
   ASSERT_TRUE(std::get<1>(range_b) == 1);
 }
 
+TEST(MemPlanning, MemReuseWithTypeCast) {
+  int M = 1024;
+  int N = 1024;
+  int K = 2048;
+
+  BufHandle AP("A", {M, K}, kFloat);
+  BufHandle BP("B", {K, N}, kFloat);
+  BufHandle EP("E", {M, N}, kQUInt8);
+  BufHandle FP("F", {M, N}, kQUInt8);
+
+  Tensor CT = Reduce(
+      "gemm",
+      {{M, "M"}, {N, "N"}},
+      Sum(),
+      [&](const ExprHandle& m, const ExprHandle& n, const ExprHandle& k) {
+        return AP.load(m, k) * BP.load(k, n);
+      },
+      {{K, "K"}});
+  Tensor DT = Compute(
+      "relu",
+      {{M, "M"}, {N, "N"}},
+      [&](const ExprHandle& m, const ExprHandle& n) {
+        auto zero = Cast::make(CT.buf()->dtype(), 0);
+        return CompareSelect::make(
+            CT.load(m, n), zero, zero, CT.load(m, n), kLT);
+      });
+
+  VarHandle i("i", kInt);
+  VarHandle j("j", kInt);
+  StorePtr store_e = Store::make(
+      EP, {i, j}, Cast::make(kQUInt8, DT.load(i, j) + DT.load(i, j)));
+  StmtPtr loop_e = For::make(i, 0, 32, For::make(j, 0, 32, store_e));
+
+  StorePtr store_f =
+      Store::make(FP, {i, j}, Load::make(EP, {i, j}) + Load::make(EP, {i, j}));
+  StmtPtr loop_f = For::make(i, 0, 32, For::make(j, 0, 32, store_f));
+  auto stmt = Block::make({CT.stmt(), DT.stmt(), loop_e, loop_f});
```

Comment:
This looks weird to mix `Compute` API with constructing IR by hand. Why can't we use `Compute` for EP and FP?

Reply:
There is a `cast` in the store statement. I tried `Compute`, and it does not generate `cast`s automatically. Will have to manually construct it.

Reply:
You don't need it to generate `cast` automatically - you specify the expression yourself there and can include `cast` too.